Fixed the issue where long values overflowed the modal in the pages u…

Annaseli opened the pull request:
…nder the "Administration" tab in the UI.

Closes #8489

## Issue
Long values overflowed modal windows across multiple pages in the "Administration" tab. Initially reported for one modal, but found in Users, Groups, and other sections (for other sections I opened another issue).

## Fix
* Applied truncation for long values in modals.
* Users can hover to see full value and copy the truncated value.
* Fix applied globally to relevant components where was possibly.

## Changes Made

### 1. "Add to Group" Pop-up (Groups -> click on some group name -> Add Members)

Updated the **DataTable** component to prevent overflow.

#### Changed in files:
webui/src/lib/components/controls.jsx (DataTable)
webui/src/lib/components/auth/forms.jsx (AttachModal)

#### The issue screenshot:
![Add_to_Group_issue](https://github.com/user-attachments/assets/e930b79b-00a1-4fbf-b254-67a969d57fb4)

#### Fix:
![Add_to_Group_fix](https://github.com/user-attachments/assets/1416d313-63c6-41b1-bbe4-5440bccca20f)


#### Note 
This pop-up is implemented using a **"DataTable"** and **AttachModal**, which is also used in other functions.

For example it fixed in addition this issue:
Issue:
![long_group_name_issue_1](https://github.com/user-attachments/assets/716f4f92-6a46-4b65-9ded-4a80e3e91237)
![long_group_name_issue_2](https://github.com/user-attachments/assets/8f2a164a-fda0-4bb8-a1dd-4d454193b649)

Fix:
![long_group_name_fix](https://github.com/user-attachments/assets/5833eb51-cd4e-4ba1-ba5e-00fb1f905595)


### 2. "Groups" Page (Groups -> Group Name) "Users" Page (Users -> User Name)

Fixed GroupHeader, PolicyHeader, UserHeader overflow.

#### Changed in files:
webui/src/lib/components/auth/nav.jsx (GroupHeader, PolicyHeader, UserHeader)
* Chaned in PolicyHeader since it has the same structure as GroupHeader and UserHeader

#### The issue screenshot:
![long_name_title_issue_1](https://github.com/user-attachments/assets/66935854-8e2f-4bb2-bb3d-7156c9a83b72)

#### Fix:
![long_name_title_fix](https://github.com/user-attachments/assets/cff55261-cb3a-4ffe-bc3b-a1c6ab1c954a)


### 3. Fixed user name overflow in confirmation pop-up (Users -> selecting a user with a long name -> Access Credentials -> Create Access Key)

#### Changed in files:
webui/src/pages/auth/users/user/credentials.jsx (getMsg function)

#### The issue screenshot:
![long_user_name_in_confirm_issue](https://github.com/user-attachments/assets/425b7075-fe26-4a30-9739-b05b15427cc6)

#### Fix:
![long_user_name_in_confirm_fix](https://github.com/user-attachments/assets/82c3c318-aa62-477a-890f-fbeaf539a541)


## Testing
Verified on lakeFS OSS using local DB & object store.

guy-har reviewed (changes requested):
Looks Amazing!
Great work! 
requesting changes because the checkbox has too much place now

![image](https://github.com/user-attachments/assets/4e3353a0-c4e1-45fb-a87f-ee19adf5c146)

itaigilo commented on webui/src/styles/globals.css:
Generally speaking, we should try to avoid using globals.css as much as possible.
Historically it's been used quite heavily, but there's a task to change this.
The reasoning behind it is to let the bootstrap framework do all the UI work for us, and "guide" it using [bootstrap classes](https://getbootstrap.com/docs/5.2/utilities/borders/). When we're using global classes, it's easy to fix one thing while breaking another one. Using the bootstrap classes gives us more confidence that the maintainers of the framework already checked this on different components.
I would start with the [table component](https://react-bootstrap.netlify.app/docs/components/table), and see which of these you can utilize.

Anyway, classes should be added to `globals.css` only as a last resort.

Annaseli replied:
Removed the classes I created here

itaigilo commented on webui/src/lib/components/controls.jsx:
Although it's tempting, please try to avoid re-formatting the whole file (unless you're changing big parts of it).
It makes reviewing harder, and messes with the git history.

There is a plan to apply auto-formatting on the React code, but it's not applied yet.

Annaseli replied:
Removed the spaces to how they were in the new PR

itaigilo commented on webui/src/pages/auth/users/user/credentials.jsx:
```diff
@@ -43,7 +44,18 @@ const UserCredentialsList = ({ userId, after, onPaginate }) => {
         );
 
-        const getMsg = (email) => <span>Create new credentials for user <strong>{email}</strong>?</span>;
+        const getMsg = (email) => (
```
Can you please explain this change?

itaigilo reviewed (changes requested):
Great first big PR - 
Really nice effort mapping the issues, looking around, and getting to a complete solution.

Some notes:
1. Agree with @guy-har 's comment about limiting the width of the checkbox.
2. You should consider splitting this PR into one with the specific changes, and one with the general ones, that will apply multiple components. Since changing global components (potentially) affects the whole app, I think it's better to have a separate PR and discussion for it.
3. Nit: better to have a shorter title for the PR, for better readability. 😊 

Anyway, nice work 💪

Annaseli commented on webui/src/lib/components/auth/forms.jsx:
```diff
@@ -9,6 +9,7 @@ import {SearchIcon} from "@primer/octicons-react";
 import {useAPI} from "../../hooks/api";
 import {Checkbox, DataTable, DebouncedFormControl, AlertError, Loading} from "../controls";
 
+import "../../../styles/globals.css"
```
you're right it works without the import

Annaseli replied:
removed it in the new PR

Annaseli commented:
I addressed all the comments:

1. removed the "import styles"
2. removed the custom styles from globals.css and used bootstrap classes instead
3. split this PR to several PRs in order to address each fix in another PR
4. Narrowed the leftmose "checkbox" column using bootstrap classes
5. used shorter commit messages for better readability

So now this PR addresses only this fix:


## Issue
Long values overflowed modal windows across tables in "Groups", "Users" and "Policies" pages in the "Administration" tab.

## Fix
1. Applied truncation for long values in modals using react bootstrap classes.
2. Users can hover to see full value and copy the truncated value.

## Changes Made
1. In the table: "Groups" main page (Administration -> Groups )
2. In the table: "Add to Group" Pop-up (Administration -> Groups -> click on some group name -> Add Members)
3. In the table: "Users" main page (Administration -> Users )
4. In the table: "Add to Groups" Pop-up (Administration -> Users -> click on some group name -> Add User to Group)
5. In similar pages in the "Policies" page 

Updated the Table under the DataTable component to prevent overflow.

Changed in file:
webui/src/lib/components/controls.jsx (DataTable)

The issue screenshots:
![Add_to_Group_issue](https://github.com/user-attachments/assets/98087855-fad5-4e09-93e4-a02700ce6b0a)
![long_group_name_issue_1](https://github.com/user-attachments/assets/6bf9abfa-6f62-4c60-a99b-fcc2982e92b7)
![long_group_name_issue_2](https://github.com/user-attachments/assets/6a276744-6d7d-4aa1-b4be-f3cbeb6e2496)

Fixes:
![Add_to_Group_fix](https://github.com/user-attachments/assets/6b0e66f9-5bc1-4353-b7ee-3c6f91cfc6d3)
![long_group_name_fix](https://github.com/user-attachments/assets/0b412273-7d98-496d-9c93-a6fd7af8f373)

## Testing
Verified on lakeFS OSS using local DB & object store & ACL.

itaigilo commented:
@Annaseli I'm not sure where it stands at the moment,
But according to your screenshots it seems that the column of the checkbox is too wide again.
Is this on purpose?

Annaseli commented:
@itaigilo 
You're right, compared to the original table with the overflow issue, the checkbox column is still too wide in the screenshots above (though narrower than in my initial solution). To shrink it further, I added "col-12" in the following line of code:
`${(firstFixedCol && i === 0) ? "col-1" : "col-12"}`
in the controls.jsx file.
When all columns except the first one are set to `"col-12"`, they cause the first column to become narrower.

Now, the column appears narrower:
![image](https://github.com/user-attachments/assets/34e69af5-82eb-43b6-b9c7-5fdd643f9378)
![image](https://github.com/user-attachments/assets/13238bc5-b333-4ce6-880b-cdbe32c31e98)
![image](https://github.com/user-attachments/assets/9e678e0d-6e5a-45a3-9b03-64e4c15a3e97)

itaigilo commented on webui/src/lib/components/controls.jsx:
```diff
@@ -335,18 +335,31 @@ export const RefreshButton = ({ onClick, size = "md", variant = "light", tooltip
+                        key={header}
+                        title={header}
+                        className={`
+                            ${(firstFixedCol && i === 0) ? "col-1" : "col-12"}
```
Well, this is a usage of bootstrap classes which sort-of work for you, but it's pretty accidental. The bootstrap framework uses 12-cols grid elements, so setting an element as `col-1` makes it take 1/12 of the width of a <Row> element, and `col-12` makes it take the whole width of it. So these classes maybe work for you in this context, but there's no guarantee it'll work on other browsers, for example.

How about doing something like:
`style={`${(firstFixedCol && i === 0) ? "width: 60px" : ""`
(Not sure about the details, but the general idea is to give an actual fixed size for the first column.)

Does this give the same result?

itaigilo replied:
(Having said that, I still think that a better overall solution is required here. Either by using a different Table package that might allow this truncate ability, or using a non-table solution for the Auth components (when thinking about it, these are not actually data tables, but UI components, and should be treated as such).
But since this requires an effort which is out-of-scope for this bug fix, it can be deferred.)

Annaseli replied:
@itaigilo 

I agree that using col-* isn't the best solution, and after I checked it a few days ago further, I found that overriding the width with a style like {{"width: X"}}, is indeed a more appropriate and common approach in React.

I understood that bootstrap’s col-* classes come from its grid system, which isn't designed to be used directly inside <th> elements in tables. They attempt to control column width and they have an effect when combined with "table-layout: fixed" as I used it. Since Bootstrap’s grid system (col-*) is meant for div-based layouts rather than tables, it's not suitable perfectly inside <th>.

I found out that a better approach for fixing column widths in tables:
To use style={{ width: "X%" }} inside <th> .
or
To use <colgroup> with { width: "X%" } .

However, when I previously suggested it, I understood that it’s not a good practice and that we should use Bootstrap classes instead. The "w-*" approach didn't work for me, but "col-*" did.

In my earlier solution (before the col-* solution) , I used a fixed width, and it worked.


So, would you recommend using style={{ width: "X%" }} instead of col-* in this case?

I believe using percentages instead of a fixed 60px width is a better approach since the browser can dynamically adjust it rather than us setting a fixed size manually.

And yes, I agree with you that a more generalized approach is needed here, and that this PR provides just a compromise.

Annaseli replied:
@itaigilo 

I updated the code to use a fixed width instead of col-*, and it works well.

Initially, I considered using a dynamic width to allow the browser to adjust the column size based on different window sizes. However, after reconsidering, I agree that a fixed width is the better approach in this case. Since the checkbox column is narrow across all screen sizes and the checkbox itself has a fixed size, it makes sense for the column width to match it.

Screenshots of the fixed size implementation:
```
style={firstFixedCol && i === 0 ? { width: "30px" } : {}}
className="text-nowrap overflow-hidden text-trunc
```

![Add_to_Group_fix_fixed_width](https://github.com/user-attachments/assets/5dec3352-df10-480b-ace7-4afe0d51393f)

![long_group_name_fix_fixed_width_users](https://github.com/user-attachments/assets/2644fe03-f34d-4f37-84f1-19c5a8c579ed)

![long_group_name_fix_fixed_width_credentials](https://github.com/user-attachments/assets/5cf54372-81e8-40b5-900e-ae74d73c9c88)

![long_group_name_fix_fixed_width_groups](https://github.com/user-attachments/assets/1254a195-564a-4828-a9f3-1d1298b93db9)

itaigilo reviewed (approved):
Yey, looking good 😄 

Good to merge!

@guy-har you're still blocking this - can you please re-review?
